diff -u -r a/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake b/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake
--- a/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake	2025-02-13 11:45:28.000000000 -0600
+++ b/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake	2025-11-13 19:57:31.987415796 -0600
@@ -145,7 +145,7 @@
 set(XCB_XCB_component_deps)
 set(XCB_COMPOSITE_component_deps XCB XFIXES)
 set(XCB_DAMAGE_component_deps XCB XFIXES)
-set(XCB_IMAGE_component_deps XCB SHM)
+set(XCB_IMAGE_component_deps XCB SHM UTIL)
 set(XCB_RENDERUTIL_component_deps XCB RENDER)
 set(XCB_XFIXES_component_deps XCB RENDER SHAPE)
 set(XCB_XVMC_component_deps XCB XV)
