diff -u -r qtbase.old/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake qtbase.new/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake
--- qtbase.old/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake	2025-11-13 19:57:31.987415796 -0600
@@ -145,7 +145,7 @@
 set(XCB_XCB_component_deps)
 set(XCB_COMPOSITE_component_deps XCB XFIXES)
 set(XCB_DAMAGE_component_deps XCB XFIXES)
-set(XCB_IMAGE_component_deps XCB SHM)
+set(XCB_IMAGE_component_deps XCB SHM UTIL)
 set(XCB_RENDERUTIL_component_deps XCB RENDER)
 set(XCB_XFIXES_component_deps XCB RENDER SHAPE)
 set(XCB_XVMC_component_deps XCB XV)
diff -u -r qtbase.old/cmake/QtBuildHelpers.cmake qtbase.new/cmake/QtBuildHelpers.cmake
--- qtbase.old/cmake/QtBuildHelpers.cmake	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/cmake/QtBuildHelpers.cmake	2025-11-15 00:50:19.933641770 -0600
@@ -414,7 +414,6 @@
     # Depends on qt_internal_set_cmake_build_type
     qt_internal_setup_cmake_config_postfix()
 
-    qt_internal_setup_position_independent_code()
     qt_internal_set_link_depends_no_shared()
     qt_internal_setup_default_install_prefix()
     qt_internal_set_qt_source_tree_var()
diff -u -r qtbase.old/src/tools/rcc/rcc.cpp qtbase.new/src/tools/rcc/rcc.cpp
--- qtbase.old/src/tools/rcc/rcc.cpp	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/src/tools/rcc/rcc.cpp	2025-11-14 00:12:48.076214506 -0600
@@ -32,7 +32,7 @@
     CONSTANT_COMPRESSLEVEL_DEFAULT = -1,
     CONSTANT_ZSTDCOMPRESSLEVEL_CHECK = 1,   // Zstd level to check if compressing is a good idea
     CONSTANT_ZSTDCOMPRESSLEVEL_STORE = 14,  // Zstd level to actually store the data
-    CONSTANT_COMPRESSTHRESHOLD_DEFAULT = 70
+    CONSTANT_COMPRESSTHRESHOLD_DEFAULT = 1
 };
 
 void RCCResourceLibrary::write(const char *str, int len)
