diff -u -r qtbase.old/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake qtbase.new/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake
--- qtbase.old/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/cmake/3rdparty/extra-cmake-modules/find-modules/FindXCB.cmake	2025-11-13 19:57:31.987415796 -0600
@@ -145,7 +145,7 @@
 set(XCB_XCB_component_deps)
 set(XCB_COMPOSITE_component_deps XCB XFIXES)
 set(XCB_DAMAGE_component_deps XCB XFIXES)
-set(XCB_IMAGE_component_deps XCB SHM)
+set(XCB_IMAGE_component_deps XCB SHM UTIL)
 set(XCB_RENDERUTIL_component_deps XCB RENDER)
 set(XCB_XFIXES_component_deps XCB RENDER SHAPE)
 set(XCB_XVMC_component_deps XCB XV)
diff -u -r qtbase.old/cmake/QtBuildHelpers.cmake qtbase.new/cmake/QtBuildHelpers.cmake
--- qtbase.old/cmake/QtBuildHelpers.cmake	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/cmake/QtBuildHelpers.cmake	2025-11-15 00:50:19.933641770 -0600
@@ -414,7 +414,6 @@
     # Depends on qt_internal_set_cmake_build_type
     qt_internal_setup_cmake_config_postfix()
 
-    qt_internal_setup_position_independent_code()
     qt_internal_set_link_depends_no_shared()
     qt_internal_setup_default_install_prefix()
     qt_internal_set_qt_source_tree_var()
diff -u -r qtbase.old/src/plugins/platforms/xcb/qxcbconnection_basic.cpp qtbase.new/src/plugins/platforms/xcb/qxcbconnection_basic.cpp
--- qtbase.old/src/plugins/platforms/xcb/qxcbconnection_basic.cpp	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/src/plugins/platforms/xcb/qxcbconnection_basic.cpp	2025-11-23 02:23:12.697507021 -0600
@@ -361,6 +361,9 @@
 
 void QXcbBasicConnection::initializeXKB()
 {
+    // Disable the xkb extension. This prevents a dependency on libx11-xkbcommon.so.
+    // Drawback: text can't be selected by shift + arrow keys
+    return;
     const xcb_query_extension_reply_t *reply = xcb_get_extension_data(m_xcbConnection, &xcb_xkb_id);
     if (!reply || !reply->present) {
         qCWarning(lcQpaXcb, "XKeyboard extension not present on the X server");
diff -u -r qtbase.old/src/plugins/platforms/xcb/qxcbconnection_basic.h qtbase.new/src/plugins/platforms/xcb/qxcbconnection_basic.h
--- qtbase.old/src/plugins/platforms/xcb/qxcbconnection_basic.h	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/src/plugins/platforms/xcb/qxcbconnection_basic.h	2025-11-23 01:31:27.807403222 -0600
@@ -48,7 +48,7 @@
     bool hasXShape() const { return m_hasXhape; }
     bool hasXRandr() const { return m_hasXRandr; }
     bool hasInputShape() const { return m_hasInputShape; }
-    bool hasXKB() const { return m_hasXkb; }
+    bool hasXKB() const { return false; }
     bool hasXRender(int major = -1, int minor = -1) const {
         if (m_hasXRender && major != -1 && minor != -1)
             return m_xrenderVersion >= std::pair(major, minor);
diff -u -r qtbase.old/src/tools/rcc/rcc.cpp qtbase.new/src/tools/rcc/rcc.cpp
--- qtbase.old/src/tools/rcc/rcc.cpp	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/src/tools/rcc/rcc.cpp	2025-11-23 02:47:34.103288435 -0600
@@ -32,7 +32,7 @@
     CONSTANT_COMPRESSLEVEL_DEFAULT = -1,
     CONSTANT_ZSTDCOMPRESSLEVEL_CHECK = 1,   // Zstd level to check if compressing is a good idea
     CONSTANT_ZSTDCOMPRESSLEVEL_STORE = 14,  // Zstd level to actually store the data
-    CONSTANT_COMPRESSTHRESHOLD_DEFAULT = 70
+    CONSTANT_COMPRESSTHRESHOLD_DEFAULT = 1 // compress if savings are at least 1%
 };
 
 void RCCResourceLibrary::write(const char *str, int len)
diff -u -r qtbase.old/src/tools/uic/CMakeLists.txt qtbase.new/src/tools/uic/CMakeLists.txt
--- qtbase.old/src/tools/uic/CMakeLists.txt	2025-02-13 11:45:28.000000000 -0600
+++ qtbase.new/src/tools/uic/CMakeLists.txt	2025-11-23 02:24:23.896939637 -0600
@@ -4,6 +4,7 @@
 #####################################################################
 ## uic Tool:
 #####################################################################
+return() # prevent build error
 
 qt_get_tool_target_name(target_name uic)
 qt_internal_add_tool(${target_name}
